### PR TITLE
GPO Management and more...

### PR DESCRIPTION
**Pull request by KnifMelti**
Just started...

### CODE REVIEW

**Comment by KnifMelti** [comment]
This optimization **if run as System** I need help with to confirm it's all good, @Romanitho!

**Comment by KnifMelti** [comment]
Remaining:
- $UpdatesAtTime (needs to be written during install to the registry - handled by Invoke-PostUpdateActions the next Release)
- $UpdatesAtTime to the Policies too
- Text changes/improvements in ADMX/ADML
- Global README.md

**Comment by KnifMelti** [comment]
Forgot the settings:
**-NoClean**
**-DesktopShortcut**
**-StartMenuShortcut**

But, I'll leave them for the time being;  maybe in a future...

**Comment by KnifMelti** [comment]
The GPO ADMX/ADML now validated: https://developer.vmware.com/samples/7115/windows-10---validate-admx-for-ingestion

**Comment by KnifMelti** [comment]
Mods (files) for Notepad++.Notepad++-(install/uninstall/preinstall etc.) can't be opened by default on from an IIS server.
Must be administrated.

**Comment by KnifMelti** [comment]
Logrotation closes #235

**Comment by KnifMelti** [comment]
@Romanitho Could you verify if I'm doing it correctly in ` Invoke-PostUpdateActions` when it comes to WAU auto updates if new version is available on Github and the registry changes for **-MaxLogFiles**/**-MaxLogSize**?

File: Winget-AutoUpdate/functions/Invoke-PostUpdateActions.ps1 [hunk 1]
```diff
@@ -43,6 +43,16 @@ function Invoke-PostUpdateActions {
...
+    #Set WAU_MaxLogFiles/WAU_MaxLogSize if not set
+    $MaxLogFiles = Get-ItemProperty $regPath -Name WAU_MaxLogFiles -ErrorAction SilentlyContinue
+    if (!$MaxLogFiles) {
+        New-ItemProperty $regPath -Name WAU_MaxLogFiles -Value 3 -PropertyType DWord -Force | Out-Null
+        New-ItemProperty $regPath -Name WAU_MaxLogSize -Value 1048576 -PropertyType DWord -Force | Out-Null
+
+        #log
+        Write-Log "-> MaxLogFiles/MaxLogSize setting was missing. Fixed with 3/1048576 (in bytes, default is 1048576 = 1 MB)."
+    }
+
```
Comment (Romanitho) [inline Winget-AutoUpdate/functions/Invoke-PostUpdateActions.ps1:55]: seems good to me :)

**Comment by Romanitho** [comment]
Really thank you for this hard work. And sorry to be less present on this project at the moment. But you are a very good captain 😉 Let me know when this part is ready and I think it will be the perfect time for a new version 😁

**Comment by KnifMelti** [comment]
I'm using it on over 200 clients and rolling out to 5-6K when I'm done with the documentation etc.
But, It's a good idea to check every bit ones again (how many times now...) to make sure there's nothing game breaking in it!
I'll hollow when I'm ready then, leaving it as ready for review.

**Comment by KnifMelti** [comment]
@Romanitho My work done!

**Comment by KnifMelti** [comment]
Really, I have to let it go...

**Comment by KnifMelti** [comment]
@Romanitho Done, done, done... 😜

File: Winget-AutoUpdate/Winget-Upgrade.ps1 [hunk 2]
```diff
@@ -14,27 +14,76 @@ $Script:IsSystem = [System.Security.Principal.WindowsIdentity]::GetCurrent().IsS
...
+            Write-Log "Activated WAU GPO Management detected, comparing..."
+            if ($null -ne $ChangedSettings -and $ChangedSettings -ne 0) {
+                Write-Log "Changed settings detected and applied" "Yellow"
+            }
+            else {
+                Write-Log "No Changed settings detected" "Yellow"
+            }
+        }
+        Write-Log "Max Log Size reached: $MaxLogSize bytes - Rotated Logs"
+    }
```
Comment (Romanitho) [inline Winget-AutoUpdate/Winget-Upgrade.ps1:74]: Quick question, from line 56 to 74, could this not be part of Invoke-LogRotation itself ? to get Winget-Upgrade.ps1 "cleaner" ?

Comment (KnifMelti) [reply Winget-AutoUpdate/Winget-Upgrade.ps1:74]: Possibly...

Comment (Romanitho) [reply Winget-AutoUpdate/Winget-Upgrade.ps1:74]: haha. just asking, not an obligation

Comment (KnifMelti) [reply Winget-AutoUpdate/Winget-Upgrade.ps1:74]: Do you want me to examine it (and possibly change)?

Comment (Romanitho) [reply Winget-AutoUpdate/Winget-Upgrade.ps1:74]: only if you have time for it. don't spend time for nothing ^^

Comment (KnifMelti) [reply Winget-AutoUpdate/Winget-Upgrade.ps1:74]: #270

**Pull request merged**